autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ScalingRequest.java

```diff
@@ -201,19 +206,21 @@ private void createHistoryAndNotify(int adjustmentCount, int totalNodes, String
                 cluster.getClusterPertain().getTenant(), System.currentTimeMillis());
     }
 
-    private String getMessageForCBStatus(String cbMessage) {
-        switch (cbMessage) {
-            case "HTTP 409 Conflict":
-                return messagesService.getMessage(MessageCode.CLUSTER_UPDATE_IN_PROGRESS, List.of(policy.getAlert().getAlertType()));
-            case "HTTP 400 Bad Request":
-            case "HTTP 500 Internal Server Error":
-            case "HTTP 404 Not Found":
-                return messagesService.getMessage(MessageCode.CLUSTER_NOT_AVAILABLE, List.of(policy.getAlert().getAlertType()));
-            case "SUCCESS":
-                return messagesService.getMessage(AUTOSCALING_ACTIVITY_SUCCESS,
-                        List.of(policy.getAlert().getAlertType(), policy.getHostGroup(), totalNodes, desiredNodeCount));
-            default:
-                return cbMessage;
+    private String getMessageForCBException(Exception cbApiException) {
+        String cbApiError = cbApiException.getMessage();
+        if (cbApiException instanceof ClientErrorException) {
+            try (Response exceptionResponse = ((ClientErrorException) cbApiException).getResponse()) {
+                cbApiError = exceptionResponse.readEntity(ExceptionResponse.class).getMessage();
+            } catch (Exception ex) {
+                LOGGER.error("Error processing CB API Exception '{}'", cbApiException, ex);
+            }
         }
+        return messagesService.getMessage(MessageCode.CLUSTER_SCALING_FAILED,
```

Comment:
Should 404 be handled explicitly? (Extremely unlikely though - since that would mean a trigger before Periscope syncs a deleted cluster?)
What happens when a cluster is in STOPPED state?

Reply:
404 need not handled explicity since it just needs to be logged similar to other error codes.
When cluster is stopped, LOAD-Based autoscaling also stops. For TIME-Based autoscaling a message is added in event history saying "Autoscaling could not be triggered because cluster was not available"